jaeger_client/config.py

```diff
@@ -303,6 +303,25 @@ def local_agent_reporting_host(self) -> str:
         else:
             return DEFAULT_REPORTING_HOST
 
+    @property
+    def local_agent_reporting_host_use_ipv6(self) -> bool:
```

Comment:
Since you're automatically determine if the address is IPv6-only, what is the value proposition of this new property? To put it differently, what is the downside of not exposing it and always handling the v4/v6 automatically?

Reply:
Resonable. Added `_create_socket` method to TUDPTransport class, that can be overriden, removed config options.

Also, socket.set_blocking() requires boolean, so i removed the following block (perhaps there is a reason to keep it):
```
if blocking:
    blocking = 1
else:
    blocking = 0
```

Reply:
Added tests